.github/workflows/simtest.yml

```diff
@@ -5,29 +5,119 @@ on:
   #   - cron: '0 3 * * *'  # every day at 2am
   workflow_dispatch:
 
+env:
+  CARGO_TERM_COLOR: always
+  # Disable incremental compilation.
+  #
+  # Incremental compilation is useful as part of an edit-build-test-edit cycle,
+  # as it lets the compiler avoid recompiling code that hasn't changed. However,
+  # on CI, we're not making small edits; we're almost always building the entire
+  # project from scratch. Thus, incremental compilation on CI actually
+  # introduces *additional* overhead to support making future builds
+  # faster...but no future builds will ever occur in any given CI environment.
+  #
+  # See https://matklad.github.io/2021/09/04/fast-rust-builds.html#ci-workflow
+  # for details.
+  CARGO_INCREMENTAL: 0
+  # Allow more retries for network requests in cargo (downloading crates) and
+  # rustup (installing toolchains). This should help to reduce flaky CI failures
+  # from transient network timeouts or other issues.
+  CARGO_NET_RETRY: 10
+  RUSTUP_MAX_RETRIES: 10
+  # Don't emit giant backtraces in the CI logs.
+  RUST_BACKTRACE: short
+
 jobs:
-  run-simtest:
-    permissions:
-      # The "id-token: write" permission is required or Machine ID will not be
-      # able to authenticate with the cluster.
-      id-token: write
-      contents: read
-    runs-on: [ self-hosted, self-hosted-arc ]
+  simtest:
+    timeout-minutes: 120
+    runs-on: [ self-hosted, simtest-01 ]
+    env:
+      RUST_LOG: off
+      # Runs tests much faster - disables signing and verification
+      USE_MOCK_CRYPTO: 1
+      MSIM_TEST_NUM: 30
+      SIM_STRESS_TEST_DURATION_SECS: 300
+      CARGO_TERM_COLOR: always
+
     steps:
-      - name: Install Teleport
-        uses: teleport-actions/setup@75c810ac63a7e9b28ca4795c6111619f2a13ac71 # pin@v1
-        with:
-          version: 11.3.1
-      - name: Authorize against Teleport
-        id: auth
-        uses: teleport-actions/auth@6e73aa03a98e781ca6692817fbb6e554640a56b3 # pin@v1
-        with:
-          # Specify the publically accessible address of your Teleport proxy.
-          proxy: proxy.mysten-int.com:443
-          # Specify the name of the join token for your bot.
-          token: sui-simtest-token
-          # Specify the length of time that the generated credentials should be
-          # valid for. This is optional and defaults to "1h"
-          certificate-ttl: 2h
+      - name: Checkout sui repo and cargo clean
+        run: |
+          cd ~/sui && git fetch origin && git rebase origin/main && cargo clean
```

Comment:
Why this approach instead of `actions/checkout`?

Reply:
Ah, good point. Left over teleport stuff. Will replace.

Reply:
We want to preserve the directory and all the work in the same location `/home/ubuntu/sui`, so that people can ssh and debug failed runs. I am pretty sure that `actions/checkout` will be much harder to stick to the same location and whipe things out.